addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3431,9 +3431,22 @@ export class OdooEditor extends EventTarget {
         if (!rangeContent.hasChildNodes()) {
             return;
         }
+        const closestLi = closestElement(selection.anchorNode, 'li');
+        const commonAncestorElement = closestElement(range.commonAncestorContainer);
+        const closestNode = closestLi || closestBlock(commonAncestorElement);
+         // Get the list of ancestor elements starting from the provided commonAncestorElement up to the closest li or block-level element.
+        const ancestorsList = commonAncestorElement && !isBlock(rangeContent.firstChild)
+                ? [commonAncestorElement, ...ancestors(commonAncestorElement, closestNode)]
+                : [];
+        // Wrap rangeContent with clones of their ancestors to keep the styles.
+        for (const ancestor of ancestorsList) {
+            const clone = ancestor.cloneNode();
+            clone.append(...rangeContent.childNodes);
+            rangeContent.appendChild(clone);
+        }
```

Comment:
You don't seem to use `ancestorsList` outside of the `for`, so there is no use for the empty array fallback. You can do:
```js
if (commonAncestorElement && !isBlock(rangeContent.firstChild)) {
    // Get the list of ancestor elements starting from the provided 
    // commonAncestorElement up to the closest li or block-level element.
    const blockOrLi = closestLi || closestBlock(commonAncestorElement);
    const ancestorsList = [commonAncestorElement, ...ancestors(commonAncestorElement, blockOrLi)];
    // Wrap rangeContent with clones of their ancestors to keep the styles.
    for (const ancestor of ancestorsList) {
        const clone = ancestor.cloneNode();
        clone.append(...rangeContent.childNodes);
        rangeContent.appendChild(clone);
    }
}
```

Reply:
Great, thanks ! 👍